Actual exception swallowed with jet submit

I had an exception occur in the pipeline building code, but got a report like this:

```
jet submit -v target/jet-job-1.0-SNAPSHOT-jar-with-dependencies.jar
Verbose mode is on, setting logging level to INFO
Submitting JAR 'target/jet-job-1.0-SNAPSHOT-jar-with-dependencies.jar' with arguments []
13:41:56.345 [ INFO] [c.h.i.c.AbstractConfigLocator] Loading configuration '/Users/mtopol/Desktop/data-engineering-talk/hazelcast-jet-4.0/config/hazelcast-client.yaml' from System property 'hazelcast.client.config'
13:41:56.360 [ INFO] [c.h.i.c.AbstractConfigLocator] Using configuration file at /Users/mtopol/Desktop/data-engineering-talk/hazelcast-jet-4.0/config/hazelcast-client.yaml
13:41:56.649 [ INFO] [c.h.c.i.s.ClientInvocationService] [4.0] Running with 2 response threads, dynamic=true
13:41:56.699 [ INFO] [c.h.c.LifecycleService] [4.0] HazelcastClient 4.0 (20200204 - 8b959f5) is STARTING
13:41:56.700 [ INFO] [c.h.c.LifecycleService] [4.0] HazelcastClient 4.0 (20200204 - 8b959f5) is STARTED
13:41:56.740 [ INFO] [c.h.c.i.c.ClientConnectionManager] [4.0] Trying to connect to cluster: jet
13:41:56.747 [ INFO] [c.h.c.i.c.ClientConnectionManager] [4.0] Trying to connect to [127.0.0.1]:5701
13:41:56.779 [ INFO] [c.h.c.LifecycleService] [4.0] HazelcastClient 4.0 (20200204 - 8b959f5) is CLIENT_CONNECTED
13:41:56.780 [ INFO] [c.h.c.i.c.ClientConnectionManager] [4.0] Authenticated with server [127.0.0.1]:5701:1a8b3822-b4e5-4e4f-8658-9daef454016c, server version: 4.0, local address: /127.0.0.1:61913
13:41:56.783 [ INFO] [c.h.i.d.Diagnostics] [4.0] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
13:41:56.787 [ INFO] [c.h.c.i.s.ClientClusterService] [4.0]
Members [1] {
	Member [127.0.0.1]:5701 - 1a8b3822-b4e5-4e4f-8658-9daef454016c
}
13:41:56.886 [ INFO] [c.h.c.i.s.ClientStatisticsService] Client statistics is enabled with period 5 seconds.
13:41:56.931 [ INFO] [c.h.c.LifecycleService] [4.0] HazelcastClient 4.0 (20200204 - 8b959f5) is SHUTTING_DOWN
13:41:56.956 [ INFO] [c.h.c.i.c.ClientConnectionManager] [4.0] Removed connection to endpoint: [127.0.0.1]:5701:1a8b3822-b4e5-4e4f-8658-9daef454016c, connection: ClientConnection{alive=false, connectionId=1, channel=NioChannel{/127.0.0.1:61913->/127.0.0.1:5701}, remoteEndpoint=[127.0.0.1]:5701, lastReadTime=2020-03-12 13:41:56.937, lastWriteTime=2020-03-12 13:41:56.937, closedTime=2020-03-12 13:41:56.932, connected server version=4.0}
13:41:56.957 [ INFO] [c.h.c.LifecycleService] [4.0] HazelcastClient 4.0 (20200204 - 8b959f5) is CLIENT_DISCONNECTED
13:41:56.961 [ INFO] [c.h.c.LifecycleService] [4.0] HazelcastClient 4.0 (20200204 - 8b959f5) is SHUTDOWN
com.hazelcast.jet.picocli.CommandLine$ExecutionException: Error while calling command (public void com.hazelcast.jet.server.JetCommandLine.submit(com.hazelcast.jet.server.JetCommandLine$Verbosity,java.lang.String,java.lang.String,java.lang.String,java.io.File,java.util.List) throws java.lang.Exception): com.hazelcast.client.HazelcastClientNotActiveException: Client is shutting down
	at com.hazelcast.jet.picocli.CommandLine.execute(CommandLine.java:1207)
	at com.hazelcast.jet.picocli.CommandLine.access$800(CommandLine.java:141)
	at com.hazelcast.jet.picocli.CommandLine$RunAll.handle(CommandLine.java:1416)
	at com.hazelcast.jet.picocli.CommandLine$RunAll.handle(CommandLine.java:1376)
	at com.hazelcast.jet.picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1243)
	at com.hazelcast.jet.picocli.CommandLine.parseWithHandlers(CommandLine.java:1526)
	at com.hazelcast.jet.server.JetCommandLine.runCommandLine(JetCommandLine.java:157)
	at com.hazelcast.jet.server.JetCommandLine.main(JetCommandLine.java:134)
Caused by: com.hazelcast.client.HazelcastClientNotActiveException: Client is shutting down
	at com.hazelcast.client.impl.spi.impl.ClientInvocation.notifyException(ClientInvocation.java:236)
	at com.hazelcast.client.impl.spi.impl.ClientInvocation.invokeOnSelection(ClientInvocation.java:173)
	at com.hazelcast.client.impl.spi.impl.ClientInvocation.invoke(ClientInvocation.java:139)
	at com.hazelcast.client.impl.spi.ProxyManager.initialize(ProxyManager.java:332)
	at com.hazelcast.client.impl.spi.ProxyManager.getOrCreateProxyInternal(ProxyManager.java:255)
	at com.hazelcast.client.impl.spi.ProxyManager.getOrCreateProxy(ProxyManager.java:224)
	at com.hazelcast.client.impl.clientside.HazelcastClientInstanceImpl.getDistributedObject(HazelcastClientInstanceImpl.java:683)
	at com.hazelcast.client.impl.clientside.HazelcastClientInstanceImpl.getDistributedObject(HazelcastClientInstanceImpl.java:677)
	at com.hazelcast.client.impl.clientside.HazelcastClientInstanceImpl.getRingbuffer(HazelcastClientInstanceImpl.java:513)
	at com.hazelcast.jet.impl.observer.ObservableImpl.destroy(ObservableImpl.java:140)
	at java.base/java.util.concurrent.ConcurrentHashMap$ValuesView.forEach(ConcurrentHashMap.java:4772)
	at com.hazelcast.jet.impl.AbstractJetInstance.shutdown(AbstractJetInstance.java:173)
	at com.hazelcast.jet.impl.JetBootstrap$InstanceProxy.shutdown(JetBootstrap.java:320)
	at com.hazelcast.jet.impl.JetBootstrap.executeJar(JetBootstrap.java:129)
	at com.hazelcast.jet.server.JetCommandLine.submit(JetCommandLine.java:209)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.hazelcast.jet.picocli.CommandLine.execute(CommandLine.java:1189)
	... 7 more
Caused by: java.io.IOException: No connection found to cluster.
	at com.hazelcast.client.impl.connection.nio.ClientConnectionManagerImpl.checkInvocationAllowed(ClientConnectionManagerImpl.java:487)
	at com.hazelcast.client.impl.spi.impl.AbstractClientInvocationService.checkInvocationAllowed(AbstractClientInvocationService.java:194)
	at com.hazelcast.client.impl.spi.impl.ClientInvocation.invokeOnSelection(ClientInvocation.java:161)
	... 25 more
```

Summary: it connected and then abruptly, with no apparent cause, disconnected and reported "No connection found to cluster."

The actual error was a `NoClassDefFoundError` due to a missing runtime dependency `hazelcast-jet-python-4.0.jar`.

--- a/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JetBootstrap.java
+++ b/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JetBootstrap.java
@@ -129,7 +129,7 @@ public final class JetBootstrap {
                 try {
                     remembered.shutdown();
                 } catch (Throwable t) {
-                    System.err.println("Client shutdown failed with:");
+                    System.err.println("Shutdown failed with:");
                     t.printStackTrace();
                 }
             }

--- a/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JetBootstrap.java
+++ b/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JetBootstrap.java
@@ -126,7 +126,12 @@ public final class JetBootstrap {
         } finally {
             JetInstance remembered = JetBootstrap.supplier.remembered();
             if (remembered != null) {
-                remembered.shutdown();
+                try {
+                    remembered.shutdown();
+                } catch (Throwable t) {
+                    System.err.println("Client shutdown failed with:");
+                    t.printStackTrace();
+                }
             }
             JetBootstrap.supplier = null;
         }

